pkg/volume/validation/pv_validation.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"github.com/golang/glog"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/volume"
```

Comment:
Just so I understand this, the pseudo code of this package will look like:

```golang
func Validation(pv *api.PersistentVolume) Errorlist {

case pv.Spec.VolumeSource.(type) {
when api.NFSSource:
  validateNFSSource(pv)
when api.EBSElasticStore:
  validateEBSSource(pv).
...


return erroList
}
```

Each plugin author can go ahead and add their own validation there and this package can be then imported from `pkg/api/validation` and chosen to validate the PV during creation.

I think we can do that, need to think bit more.